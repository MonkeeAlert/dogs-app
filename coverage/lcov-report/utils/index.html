
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.73% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>126/142</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.59% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>54/67</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.47% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>19/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.23% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>120/136</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="constants.ts"><a href="constants.ts.html">constants.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="functions.ts"><a href="functions.ts.html">functions.ts</a></td>
	<td data-value="92.62" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.62" class="pct high">92.62%</td>
	<td data-value="122" class="abs high">113/122</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="63" class="abs high">54/63</td>
	<td data-value="94.73" class="pct high">94.73%</td>
	<td data-value="19" class="abs high">18/19</td>
	<td data-value="92.24" class="pct high">92.24%</td>
	<td data-value="116" class="abs high">107/116</td>
	</tr>

<tr>
	<td class="file low" data-value="tinyUrlHelper.ts"><a href="tinyUrlHelper.ts.html">tinyUrlHelper.ts</a></td>
	<td data-value="41.66" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 41%"></div><div class="cover-empty" style="width: 59%"></div></div>
	</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	</tr>

<tr>
	<td class="file empty" data-value="types.ts"><a href="types.ts.html">types.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Fri Apr 08 2022 23:23:18 GMT+0300 (Eastern European Summer Time)
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    