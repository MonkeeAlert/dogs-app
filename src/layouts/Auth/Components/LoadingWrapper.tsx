import React, {ReactElement, useState} from 'react';
import {ActivityIndicator, StyleSheet, View} from 'react-native';
import {useSelector} from 'react-redux';
import {useTheme} from '../../../assets/theme';
import {getUserStorage} from '../../../redux/rootSelector';

interface Props {
  children: ReactElement | ReactElement[];
}

export const LoadingWrapper: React.FC<Props> = ({children}) => {
  const {mode} = useTheme();
  const [height, setHeight] = useState(0);

  const {user} = useSelector(getUserStorage);

  return (
    <View
      style={[styles.container, height > 0 ? {height} : null]}
      onLayout={e => setHeight(e.nativeEvent.layout.height)}>
      {user.loading ? (
        <ActivityIndicator size={'small'} color={mode.background} />
      ) : (
        children
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 21,
    marginHorizontal: 21,
  },
});
