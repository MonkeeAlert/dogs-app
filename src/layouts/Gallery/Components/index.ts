export * from './ListHeader';
export * from './SeeMore';
