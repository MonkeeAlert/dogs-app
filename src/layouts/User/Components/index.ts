export * from './BookmarksLink';
export * from './CacheLink';
export * from './List';
