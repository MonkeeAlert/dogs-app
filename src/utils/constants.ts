export const colors = {
  white: '#fffafb',
  lightGray: '#E3F1EA',
  turquoise: '#2BD9A5',
  darkTurquoise: '#2E8A6E',
  darkGray: '#2b2c28',
  black: '#131515',
};

export const text = {
  xs: 7,
  s: 14,
  m: 21,
  l: 28,
  xl: 35,
  xxl: 42,
};

export const animationConfig = {
  duration: 200,
};

export const turquoiseGradientArray = [colors.turquoise, '#2bd9a5aa'];
