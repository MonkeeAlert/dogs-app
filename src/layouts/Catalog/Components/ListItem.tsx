import {
  Dimensions,
  Image,
  Pressable,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import React from 'react';
import {Icon} from 'react-native-elements';
import {useSelector} from 'react-redux';
import {getDogsCatalog} from '../../../redux/rootSelector';
import {colors} from '../../../utils/constants';

interface Props {
  uri: string;
  idx: number;
}

const ICON_SIZE = 42;

// activeName: 'bookmarks',
// defaultName: 'bookmarks-outline',

const ListItem = ({uri, idx}: Props) => {
  const {bookmarks} = useSelector(getDogsCatalog);

  const isBookmarked = bookmarks.indexOf(uri) !== -1;

  return (
    <View
      style={[
        styles.container,
        {
          position: 'relative',
          top: idx % 2 !== 0 ? 28 : 0,
        },
      ]}>
      <Pressable>
        <Icon
          type={'ionicon'}
          name={`bookmarks`}
          color={isBookmarked ? colors.white : 'transparent'}
          containerStyle={styles.icon}
        />
      </Pressable>

      <Image source={{uri}} style={styles.image} />
    </View>
  );
};

export default ListItem;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginVertical: 7,
    marginHorizontal: 7,
  },
  image: {
    width: '100%',
    maxWidth: Dimensions.get('screen').width / 2 - 7 * 2 - 7,
    height: 200,
    borderRadius: 20,
    zIndex: -1,
  },
  icon: {
    position: 'absolute',
    top: 7,
    right: 7,
    zIndex: 1,
    elevation: 1,
    width: ICON_SIZE,
    height: ICON_SIZE,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
