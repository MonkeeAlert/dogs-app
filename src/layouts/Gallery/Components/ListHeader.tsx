import {View, Text, StyleSheet} from 'react-native';
import React, {useRef} from 'react';
import {colors, text} from '../../../utils/constants';
import ShareButton from '../../../components/buttons/ShareButton';
import DefaultButton from '../../../components/buttons/DefaultButton';
import {useDispatch, useSelector} from 'react-redux';
import {getDogsCatalog} from '../../../redux/rootSelector';
import {saveToBookmarks} from '../../../redux/actions/listActions';
import {Icon} from 'react-native-elements/dist/icons/Icon';
import {MainStyles} from '../../../assets/styles/MainStyles';
import {getBreed} from '../../../utils/functions';

interface Props {
  uri: string;
}

export const ListHeader = ({uri}: Props) => {
  const {bookmarks} = useSelector(getDogsCatalog);
  const dispatch = useDispatch();

  const handleToogleBookmark = () => dispatch(saveToBookmarks(uri));

  const isBookmark = bookmarks.indexOf(uri) !== -1;

  return (
    <View style={styles.container}>
      <Text style={styles.headerText}>More {getBreed(uri)}</Text>

      <View style={styles.headerButtons}>
        <DefaultButton
          onPress={handleToogleBookmark}
          color={isBookmark ? colors.turquoise : colors.lightGray}>
          <Icon
            testID={'ListHeader_Icon'}
            type={'ionicon'}
            name={`bookmarks`}
            color={isBookmark ? colors.white : colors.turquoise}
            size={18}
          />
        </DefaultButton>
        <ShareButton uri={uri} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    ...MainStyles.rowFull,
    borderRadius: 14,
    backgroundColor: colors.white,
    paddingLeft: 14,
    paddingRight: 11.5,
    paddingVertical: 7,
  },
  headerText: {
    fontSize: text.m,
    fontWeight: '900',
    paddingVertical: 14,
    color: colors.darkGray,
  },
  headerButtons: {
    flexDirection: 'row',
  },
});
