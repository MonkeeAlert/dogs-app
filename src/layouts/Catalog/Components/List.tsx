import {
  View,
  Text,
  FlatList,
  Image,
  StyleSheet,
  TouchableHighlight,
} from 'react-native';
import React from 'react';
import ListItem from './ListItem';

interface Props {
  data: string[];
}

const renderItem = (uri: string, idx: number) => {
  return <ListItem uri={uri} idx={idx} />;
};

const List = ({data}: Props) => {
  return (
    <FlatList
      data={data}
      numColumns={2}
      showsVerticalScrollIndicator={false}
      keyExtractor={(_, idx) => `Dog_${idx}`}
      contentContainerStyle={{paddingBottom: 21}}
      renderItem={({item, index}) => renderItem(item, index)}
    />
  );
};

export default List;
