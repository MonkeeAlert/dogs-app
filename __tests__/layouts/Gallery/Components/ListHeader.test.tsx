import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {create} from 'react-test-renderer';
import {mockedDispatch} from '../../../../jest.setup';
import {ListHeader} from '../../../../src/layouts/Gallery/Components/ListHeader';
import {colors} from '../../../../src/utils/constants';

const LIST = [
  'https://images.dog.ceo/breeds/labrador/n02099712_5648.jpg',
  'https://images.dog.ceo/breeds/retriever-golden/n02099601_8005.jpg',
  'https://images.dog.ceo/breeds/pyrenees/n02111500_7483.jpg',
];

describe('GalleryModal (ListHeader)', () => {
  beforeAll(() => {
    useSelector.mockReturnValue({
      bookmarks: LIST,
    });
    useDispatch.mockReturnValue(mockedDispatch);
  });

  it('matches snapshot', () => {
    const tree = create(
      <ListHeader uri={LIST[1].replace('golden', 'flatcoated')} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('sets specified color to icon when uri is bookmarked', () => {
    const {getAllByTestId} = render(
      <View>
        <ListHeader uri={LIST[0]} />
        <ListHeader uri={LIST[1].replace('golden', 'flatcoated')} />
      </View>,
    );

    const view = getAllByTestId('ListHeader_Icon');

    expect(view[0].children[0].props.children.props.color).toBe(colors.white);
    expect(view[1].children[0].props.children.props.color).toBe(
      colors.turquoise,
    );
  });

  it('dispatches action', () => {
    const {getAllByTestId} = render(<ListHeader uri={LIST[0]} />);
    const btn = getAllByTestId('DefaultButton_TouchableOpacity');

    act(() => {
      fireEvent.press(btn[0]);
      expect(mockedDispatch).toHaveBeenCalled();
    });
  });
});
