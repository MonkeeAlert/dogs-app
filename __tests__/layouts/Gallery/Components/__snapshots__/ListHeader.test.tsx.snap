// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryModal (ListHeader) matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fffafb",
      "borderRadius": 14,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingLeft": 14,
      "paddingRight": 11.5,
      "paddingVertical": 7,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2b2c2c",
        "fontSize": 21,
        "fontWeight": "900",
        "paddingVertical": 14,
      }
    }
  >
    More 
    retrievers
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="DefaultButton_TouchableOpacity"
    >
      <View
        style={
          Object {
            "backgroundColor": "#ededed",
            "borderRadius": 7,
            "margin": 3.5,
            "padding": 10,
          }
        }
        testID="DefaultButton_View"
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            testID="ListHeader_Icon"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#2BD9A5",
                      "fontSize": 18,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="DefaultButton_TouchableOpacity"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2BD9A5",
            "borderRadius": 7,
            "margin": 3.5,
            "padding": 10,
          }
        }
        testID="DefaultButton_View"
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#fffafb",
                      "fontSize": 18,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
